### ISSUE TITLE
Type issue with TS 4.2

### ISSUE THREAD

**Issue by nstepien** [formatjs/formatjs#2616]
## Which package?
`@formatjs/ecma402-abstract` via `react-intl`

## Describe the bug
With TypeScript 4.2 (currently in release candidate), there is a type issue:
```
node_modules/@formatjs/ecma402-abstract/types/date-time.d.ts:1:18 - error TS2430: Interface 'import("D:/repos/MediaModule/MediaModule.Web/client/node_modules/@formatjs/ecma402-abstract/types/date-time").DateTimeFormatOptions' incorrectly extends interface 'Intl.DateTimeFormatOptions'.
  Types of property 'calendar' are incompatible.
    Type '"buddhist" | "chinese" | "coptic" | "ethiopia" | "ethiopic" | "gregory" | "hebrew" | "indian" | "islamic" | "iso8601" | "japanese" | "persian" | "roc" | undefined' is not assignable to type 'Calendar | undefined'.
      Type '"ethiopia"' is not assignable to type 'Calendar | undefined'.

1 export interface DateTimeFormatOptions extends Intl.DateTimeFormatOptions {
                   ~~~~~~~~~~~~~~~~~~~~~
```

I think you want to use `'ethiopic-amete-alem' | 'ethioaa'` instead of `'ethiopia'`

![image](https://user-images.githubusercontent.com/567105/108222693-ea683980-7130-11eb-80d3-7d2b678f1bb0.png)

### Codesandbox URL

Please include a codesandbox URL if possible

### Reproducible Steps/Repo

1. Install `npm i react-intl typescript@rc`
2. import react-intl in some code
3. run `tsc`

## Expected behavior
No type issue with the upcoming TS release.

## Screenshots
N/A

## Desktop (please complete the following information):

- OS: N/A
- Browser N/A
- Version N/A

## Smartphone (please complete the following information):

- Device: N/A
- OS: N/A
- Browser N/A
- Version N/A

## Additional context
https://github.com/microsoft/TypeScript/issues/41601

**Comment by kachkaev** [comment]
Thanks for reporting this issue @nstepien! I hope that #2618 will resolve it. Typescript 4.2 final is [scheduled for Feb 23](https://github.com/microsoft/TypeScript/issues/41601) and it would be great to test a new patch release of `@formatjs/ecma402-abstract` before then 🙏

Given that `@formatjs/ecma402-abstract` is downloaded 600K times a week and TS has 17M weekly downloads, there’ll be a storm of comments if we are late 😅

**Comment by longlho** [comment]
This is a upstream TypeScript issue. I've posted a comment there.

**Comment by nstepien** [comment]
There is a new(?) issue now, where `numberingSystem` in `DateTimeFormatOption` is `string | undefined`, and not assignable to `NumberingSystem | undefined`.

**Comment by Pantalaimon** [comment]
After running a `yarn upgrade` I also got hit by incompatible types
```
error TS2430: Interface 'import("node_modules/@formatjs/ecma402-abstract/types/date-time").DateTimeFormatOp
tions' incorrectly extends interface 'Intl.DateTimeFormatOptions'.
  Types of property 'fractionalSecondDigits' are incompatible.
    Type 'number | undefined' is not assignable to type '0 | 2 | 1 | 3 | undefined'.
      Type 'number' is not assignable to type '0 | 2 | 1 | 3 | undefined'.
```
Temporarily fixed by reverting typescript to 4.1.5

**Comment by longlho** [comment]
ok this is another upstream TS issue. Will put a PR upstream

Update: https://github.com/microsoft/TypeScript/pull/42945
